app/assets/javascripts/modules/AssignedToFilter.js

```diff
@@ -9,29 +9,39 @@ moj.Modules.AssignedToFilter = {
           $('.js-all-business-units').show();
           module.getMatchingBusinessUnits(this.value);
         } else {
-          $('.js-all-business-units').hide();
+          $('.js-all-business-units, .js-no-results').hide();
+
         }
       }).trigger('keyup')
     },
 
 
-    getMatchingBusinessUnits: function (searchFor){
+    getMatchingBusinessUnits: function (searchFor) {
       $('.js-all-business-units').find('.multiple-choice')
-        .each(function( index, element ) {
-          var $elem = $(this);
-          // Remove any character that is not alphanumeric
-          var labelText = $elem.text().replace(/[^a-z0-9+]+/gi, '');
-          // Remove any character that is not alphanumeric
-          var cleanedTerm = searchFor.replace(/[^a-z0-9+]+/gi, '');
-
-          // if checkbox label test contains any part of the search term
-          if (labelText.search(cleanedTerm, 'gi') > -1){
-              $elem.show();
-          }else{
-              $elem.hide();
-              $elem.find(':checkbox').prop('checked', false);
-          }
-        })
+      .each(function (index, element) {
+        var $elem = $(this);
+        // Remove any character that is not alphanumeric
+        var labelText = $elem.text().replace(/[^a-z0-9+]+/gi, '');
+        // Remove any character that is not alphanumeric
+        var cleanedTerm = searchFor.replace(/[^a-z0-9+]+/gi, '');
+        var check = new RegExp(cleanedTerm, 'gi');
+
+
+        // if checkbox label test contains any part of the search term
+        if (check.test(labelText)) {
+          $elem.show();
+        } else {
+          $elem.hide();
+          $elem.find(':checkbox').prop('checked', false);
```

Comment:
Shouldn't all the business units that have been previously selected be replayed?

Reply:
As per Slack discussion, we'll leave this for a later improvement and just get this out for now.